### PR TITLE
393 sim use contract abi to generate methods and parameters

### LINKED ISSUE #393: SIM-Use contract abi to generate methods and parameters

### PR DESCRIPTION
**Pull request by denishacquin**
Fixes #393 

# What

- Fix mapping order of method params

# Why

- Because the params where getting mixed up

# Testing done

- tested the bug fix

# Decisions made

- used **abi** property instead of **methods** property in methods and constructors
- removed **method** property from backend response

# Checks

- [x] I have tested this code
- [x] I have reviewed my own PR
- [x] I have created an issue for this PR
- [x] I have set the PR name to the issue name

### CODE REVIEW

**Review by cristiam86 — COMMENTED** [review]
This is great, but please try also to remove the part that generates the now unused method in the backend: https://github.com/yeagerai/genlayer-simulator/blob/main/backend/node/genvm/base.py#L223

File: backend/node/genvm/base.py [hunk 1]
```diff
@@ -271,7 +271,6 @@ def get_contract_schema(contract_code: str) -> dict:
 
             contract_schema = {
                 "class": class_name,
-                "methods": methods,
```
[the remark this answers is not in the document]
Comment (denishacquin) [reply backend/node/genvm/base.py:274]: I thought so too initially but no; at line 270 we use the methods to build the abi:

`abi = GenVM.generate_abi_from_schema_methods(methods)`

[the remark this answers is not in the document]
Comment (cristiam86) [reply backend/node/genvm/base.py:274]: Alright! but we probably don't need to return it in this method

File: backend/node/genvm/base.py [hunk 2]
```diff
@@ -271,7 +271,6 @@ def get_contract_schema(contract_code: str) -> dict:
...
                 "variables": variables,
```
[the remark this answers is not in the document]
Comment (denishacquin) [reply backend/node/genvm/base.py:274]: couldn't find any reference in the frontend, removing

File: backend/node/genvm/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (denishacquin) [reply backend/node/genvm/base.py:274]: I'm confused. Is it currently OK or not? I already removed it from the returned payload.

[the remark this answers is not in the document]
Comment (cristiam86) [reply backend/node/genvm/base.py:274]: Now it seems that it is ok

**Review by cristiam86 — APPROVED** [review]

**Pull request merged**